src/Microsoft.DotNet.CodeFormatting/Analyzers/ExplicitVariableTypeAnalyzer.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Diagnostics;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.CodeFormatting.Analyzers
+{
+    [Export(typeof(DiagnosticAnalyzer))]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    class ExplicitVariableTypeAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "DNS0099";
```

Comment:
Just thought you might have those reserved for existing rules. Should I take 0003?